### PR TITLE
dbt-postgres 'microbatch' strategy

### PR DESCRIPTION
**Pull request by MichelleArk**
resolves https://github.com/dbt-labs/dbt-postgres/issues/149
[docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose) dbt-labs/docs.getdbt.com/#

### Problem

As part of https://github.com/dbt-labs/dbt-core/issues/10624, dbt-postgres needs to implement a default microbatch strategy.

### Solution

Although it is not write-optimized for partitioned data, we will use merge under-the-hood in dbt-postgres for now, as dbt-postgres does not yet support partitioning.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-postgres/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -54,10 +54,10 @@ path = "dbt/adapters/postgres/__version__.py"
 
 [tool.hatch.envs.default]
 dependencies = [
-    "dbt-adapters @ git+https://github.com/dbt-labs/dbt-adapters.git",
+    "dbt-adapters @ git+https://github.com/dbt-labs/dbt-adapters.git@base-microbatch-tests",
```
Comment (QMalcolm) [inline pyproject.toml:57]: I assume at this point we're just waiting for https://github.com/dbt-labs/dbt-adapters/pull/300 to then get this line updated. Is that correct?

Comment (MichelleArk) [reply pyproject.toml:57]: yep!

**Review by mikealfare — APPROVED** [review]
Approval conditional on repointing to `main` of course.

**Comment by MichelleArk** [comment]
Confirmed the failing tests are expected on `tests/functional/adapter/test_incremental_microbatch.py::TestPostgresMicrobatch::test_run_with_event_time`. Going to go ahead and merge on failing CI here as this will just be intermittent until https://github.com/dbt-labs/dbt-core/issues/10624 is merged.

**Pull request merged**